src/infoview.ts

```diff
@@ -1,185 +1,216 @@
-export class InfoProvider implements TextDocumentContentProvider, Disposable {
-    scheme = 'lean-info';
+h1
+{
+  display: block;
+  width: 100%;
 
-    documentSelector: DocumentSelector = {
-        scheme: this.scheme,
-    };
+  margin-bottom: 1ex;
+  margin-top: 1ex;
+  border-bottom: 1px solid black;
+  padding: 0px;
 
-    private leanInfoPath = '.lean-info';
-    leanInfoUrl = Uri.parse(`${this.scheme}:${this.leanInfoPath}`);
+  font-weight: normal;
+  font-size: 100%;
+}
 
-    private changedEmitter = new EventEmitter<Uri>();
-    onDidChange = this.changedEmitter.event;
-    private documents = new Map<string, InfoDocument>();
-    private subscriptions: Disposable[] = [];
+div.error h1 {
+  color: red;
+}
 
-    constructor(private server: Server, private leanDocs: DocumentSelector) {
-        this.subscriptions.push(workspace.onDidCloseTextDocument((doc) => {
-            const uri = doc.uri.toString();
-            if (this.documents.has(uri)) {
-                this.documents.get(uri).dispose();
-                this.documents.delete(uri);
-            }
-        }));
+div.warning h1 {
+  color: darkorange;
+}
+
+div.information h1 {
+  color: darkgreen;
+}
+
+#messages div {
+  border-top: 5px solid transparent;
+}
+
+.vscode-light h1 {
+  border-bottom: 1px solid black;
+}
+
+.vscode-dark h1 {
+  border-bottom: 1px solid white;
+}
+
+.vscode-dark div.highlight {
+  background: darkolivegreen;
+}
+
+            </style>
+            </head>
+            <body>` +
+                this.renderGoal() +
+                `<div id="messages">` + this.renderMessages() + `</div>`+
+            `</body></html>`;
     }
 
-    dispose() {
-        this.documents.forEach((v) => v.dispose());
-        for (const s of this.subscriptions) s.dispose();
+    private renderGoal() {
+        const goalState = this.curGoalState;
+        if (!goalState) return '';
+        return `<div id="goal"><h1>Tactic State</h1><pre>${goalState}</pre></div>`;
     }
 
-    provideTextDocumentContent(uri: Uri, token: CancellationToken): string {
-        const uriString = uri.toString();
-        if (!this.documents.has(uriString)) {
-            if (this.leanInfoUrl.toString() !== uriString) {
-                throw new Error(`unsupported uri: ${uri}`);
+    private renderMessages() {
+        if (!this.curFileName) return ``;
+        return this.curMessages.map((m) => {
+            const i = m.text.indexOf("\n");
+            if (i == -1) {
+                return `<div class="message ${m.severity}"><h1>${m.severity}: ${m.caption}</h1><pre>${m.text}</pre></div>`
             }
-            const doc = new InfoDocument(this.server, this.leanDocs);
-            doc.onChanged(() => this.changedEmitter.fire(uri));
-            this.documents.set(uriString, doc);
-        }
-        return this.documents.get(uriString).text;
+            return `<div class="message ${m.severity}">
+                <h1>${m.severity}: ${m.caption} ${m.text.substring(0, i)}</h1>
```

Comment:
The `m.text.substring` causes the first line of the trace output to be displayed as part of the caption.  Is there any situation where we might actually want that?